src/engine/IndexScan.h

```diff
@@ -91,6 +91,12 @@ class IndexScan final : public Operation {
     return numVariables() == target;
   }
 
+  // Full index scans will never be able to fit in the cache on datasets the
+  // size of wikidata, so we don't even need to try and waste performance.
+  [[nodiscard]] bool unlikelyToFitInCache() const override {
+    return numVariables_ == 3;
```

Comment:
This function should be passed a `MemorySize` and then you can do the generic comparison between the memorySize and the `getSizeEstimate()`.
This can even be the generic default implementation (maybe with some slack) in `Operation.h`.

Reply:
the memory size is of course the maximal cache size for lazy results.

Reply:
How accurate is the size estimate usually? I intended this interface to indicate that the result is conceptionally too large, and if that's not possible to safely assume then let the mechanism try it until it runs into the limit or it doesn't in which case it would be cached.